tokio/src/sync/watch.rs

```diff
@@ -610,6 +610,55 @@ impl<T> Receiver<T> {
         }
     }
 
+    /// This function is similar to [`Receiver::changed()`], but it takes a closure that is
+    /// called with a reference to the new value. If the closure returns `true`,
+    /// then the function returns immediately. Otherwise, it waits for a new
+    /// value and calls the closure again.
```

Comment:
All docs should start with a short single-line description of what the method does.

Reply:
Updated in cc3830ec does it look ok?

Reply:
It only works with simple closures? Not complicated ones?

I suggest phrasing it like this:

> Waits until the provided closure returns true.

Reply:
updated to work for complicated closures too 😅